subsys/net/ip/Kconfig

```diff
@@ -899,6 +899,14 @@ config NET_DISABLE_ICMP_DESTINATION_UNREACHABLE
 	  Suppress the generation of ICMP destination unreachable errors
 	  when ports that are not in a listening state receive packets.
 
+config NET_FORCE_ICMP_CHECKSUM
```

Comment:
The issue with such approach is as soon as you mix 2 hardware that have different support, i.e. one which has full cksum support, the other only partially . it will unilaterally shut the capabilities of the first one. 

Perhaps it is time to consider a finer ETHERNET_HW_<TX/RX>_CHKSUM_OFFLOAD set of options: one for ipv4/6, icmp4/6?

Reply:
I would be happy to implement either of these options and submit a new PR, but I am not sure what the best solution is.

Reply:
Perhaps we could leave capabilities as they are, but allow to provide more detailed information about checksum offloading with `get_config()` and a new enum indeed? That should have minimal impact on existing drivers, as in case the more detailed information is not available, we could just retain the current behavior (all protos offloaded).

@tbursztyka WDYT?

Reply:
Sure. The hw enum would be then for a generic support, and the config one for detailed ones. Fine by me

Reply:
Sounds good, I will implement this accordingly and update this PR.

Reply:
I have added the corresponding new flags. If you are happy with the changes, I will implement the new flag for the existing in-tree ethernet drivers. As it stands now, they would otherwise default to computing all checksums in software.